### DISCUSSION TITLE
DAP Restart request

### DISCUSSION THREAD

**Discussion by Davincible** [go-delve/delve#3049] [General]
I read that the restart request is [not supported](https://github.com/go-delve/delve/tree/master/Documentation/api/dap#disconnect-and-shutdown) yet. As I couldn't find any open issues/discussions about it I was wondering whether there are any plans to implement it

**Comment by aarzilli** [comment]
cc @suzmue @polinasok @hyangah

**Reply by suzmue** [reply]
I looked around a bit and was unable to find an issue for this either, but as far as I know there are no plans to implement the restart request right now. To restart you can always just shut down the debug server and start a new one with the same launch configuration.

**Reply by ldelossa** [reply]
There are cases where a restart is very useful. 

For instance, if you're using the debugger for local development. 

You can setup Delve inside a container which has all the dependencies + your source code mounted in from the host.
You can instruct Delve to compile the source on client connect. 
You can then debug the code and also make changes to the source code in real time. 
You can then finally tell Delve to "Restart" where it will recompile your source code. 

Delve is basically acting like a execution host in this case, with debugger functionality which can be super useful.
